### PR TITLE
Soluttion

### PR DESCRIPTION
**Pull request by denysbelei9**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,5 +1,58 @@
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:2]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

File: app/main.py [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:26]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

File: app/main.py [hunk 3]
```diff
@@ -1,5 +1,58 @@
...
+        self.coords[2] += step
+
+    def go_down(self, step: int = 1) -> None:
+        self.coords[2] -= step
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(
+            self, name: str, weight: int,
+            coords: list = None, max_load_weight: int = 0,
```
Comment (Bohdan-Salamakha) [inline app/main.py:41]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]

use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

File: app/main.py [hunk 4]
```diff
@@ -1,5 +1,58 @@
...
+            current_load: int = None
```
Comment (Bohdan-Salamakha) [inline app/main.py:42]: use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

File: app/main.py [hunk 5]
```diff
@@ -1,5 +1,58 @@
...
+    ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: list) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py:48]: cargo isn't a list

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

File: app/main.py (hunk 4 above)
Comment (denysbelei9) [reply app/main.py:42]: Done

**Pull request closed without merging**